lib/spack/spack/cmd/external.py

```diff
@@ -0,0 +1,144 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from __future__ import print_function
+from collections import defaultdict
+import os
+import re
+
+import spack
+import llnl.util.tty as tty
+
+description = "add external packages to Spack configuration"
+section = "config"
+level = "short"
+
+
+def setup_parser(subparser):
+    sp = subparser.add_subparsers(
+        metavar='SUBCOMMAND', dest='external_command')
+
+    # Find
+    sp.add_parser('find', help=external_find.__doc__)
+
+
+def _get_system_executables():
+    path = os.getenv('PATH')
+    search_paths = path.split(os.pathsep)
+    exe_to_path = {}
+    # Reverse order of search directories so that an exe in the first PATH
+    # entry overrides later entries
+    for search_path in reversed(search_paths):
```

Comment:
```suggestion
    for search_path in filter(os.path.exists, reversed(search_paths)):
```
The current code errors out if there's a path in `PATH` that does not exist

Reply:
Done